test/extensions/filters/listener/common/fuzz/listener_filter_fuzzer.cc

```diff
@@ -83,12 +87,26 @@ FuzzedHeader::FuzzedHeader(const test::extensions::filters::listener::FilterFuzz
   }
 }
 
-Api::SysCallSizeResult FuzzedHeader::next(void* buffer, size_t length) {
-  if (done()) {           // End of stream reached
-    nread_ = nreads_ - 1; // Decrement to avoid out-of-range for last recv() call
+void FuzzedHeader::next() {
+  if (!done()) {
+    nread_++;
   }
-  memcpy(buffer, data_.data(), std::min(indices_[nread_], length));
-  return Api::SysCallSizeResult{static_cast<ssize_t>(indices_[nread_++]), 0};
+}
+
+Api::SysCallSizeResult FuzzedHeader::read(void* buffer, size_t length, int flags) {
```

Comment:
Using `flags` makes for a leaky abstraction since the code here only checks for MSG_PEEK, and checks for equality instead of using a bitmask. Consider instead using a bool "peek", or even better, an enum with two states. That will make it more obvious at the call site what the intent is, at the cost of a little extra code there.